src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -483,9 +446,12 @@ private async Task<string> GetOnlinePackagePath(SdkFeatureBand sdkFeatureBand, M
         private string GetOfflinePackagePath(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, DirectoryPath? offlineCache = null)
         {
             string packagePath = Directory.GetFiles(offlineCache.Value.Value)
-                .Where(path => path.EndsWith(".nupkg"))
                 .Where(path =>
                 {
+                    if (!path.EndsWith(".nupkg"))
+                    {
+                        return false;
```

Comment:
The code in this lambda expression already called `return` 3 lines below this. `return` in a lambda returns a result, like in a function or method. The LINQ method takes a delegate that returns a boolean. This logic is identical to the original `.Where(path => path.EndsWith(".nupkg"))` except it won't allocate all the LINQ expression boilerplate to combine the `Where`s together. It is generally bad practice to use multiple `Where` expressions in a row since you can just make 1 lambda handle all the boolean logic at the same time.

Reply:
I've removed this method and merged the 1 statement into the only method calling it. I also merged the boolean logic into 1 expression with an `&&`, which is the much better way to do this.

Reply:
I see, that makes sense -- thank you for the explanation :)